exporters/export_formatter/avro_export_formatter.py

```diff
@@ -0,0 +1,68 @@
+import json
+import six
+from fastavro.writer import Writer
+import io
+
+from exporters.exceptions import ConfigurationError
+from exporters.export_formatter.base_export_formatter import BaseExportFormatter
+
+
+class AvroExportFormatter(BaseExportFormatter):
+    """
+    This export formatter provides a way of exporting items in AVRO format.
+    follows the avro schema 1.8 https://avro.apache.org/docs/1.8.1/spec.html
+    We write the items on a 1 per block way.
+    This allows for streaming and keeps the files consistency even if the process
+    ends unexpectedly
+
+        - schema(string)
+            json-encoded schema for the data to use in avro formatter
+
+        - schema_path(string)
+            path to the json schema for the data to use in avro formatter.
+            Useful for large schemas if readibility of the job json file is important
+    """
+
+    supported_options = {
+        'schema_path': {'type': six.string_types, 'default': None},
+        'schema': {'type': six.string_types, 'default': None},
+    }
+    file_extension = 'avro'
+    item_separator = ''
+
+    def __init__(self, *args, **kwargs):
+        super(AvroExportFormatter, self).__init__(*args, **kwargs)
+        self.schema = self._get_schema()
+        self._setup_writer()
+
+    def _get_schema(self):
+        if self.read_option('schema_path'):
+            with open(self.read_option('schema_path')) as fname:
+                return json.load(fname)
+        elif self.read_option('schema'):
+            return json.loads(self.read_option('schema'))
+        else:
+            raise ConfigurationError('Avro formatter requires at least one of: schema or schema_path')
+
+    def _setup_writer(self):
+        header_buffer = io.BytesIO()
+        self.writer = Writer(fo=header_buffer, schema=self.schema, sync_interval=0)
+        self.header_value = header_buffer.getvalue()
+        self._clear_buffer()
+
+    def format_header(self):
+        return self.header_value
+
+    def _clear_buffer(self):
+        self.writer.fo.seek(0)
+        self.writer.fo.truncate()
+
+    def format(self, item):
+        try:
+            self.writer.write(item)
+            item_value = self.writer.fo.getvalue()
+            return item_value
+        except Exception as e:
+            print(e)
```

Comment:
This looks suspicious, all exceptions are being swallowed.
Why is this needed?
If you really need this, wouldn't it be best to specify the base exceptions that may happen?

Reply:
Aaand this is code that shouldnt be there, as I was using it for debugging purposes. Given how avro format is defined  and how items are streamed there is no reason to have this in the library.